pkg/oci/remote/signatures.go

```diff
@@ -49,6 +52,42 @@ func Signatures(ref name.Reference, opts ...Option) (oci.Signatures, error) {
 	}, nil
 }
 
+func Bundle(ref name.Reference, opts ...Option) (*sgbundle.Bundle, error) {
+	o := makeOptions(ref.Context(), opts...)
+	img, err := remoteImage(ref, o.ROpt...)
+	if err != nil {
+		return nil, err
+	}
+	// TODO: We can check for a specific predicate type here using
+	// img.Manifest() and looking at the annotations. To do so, we would need
+	// to thread the CLI flag through to here.
+	layers, err := img.Layers()
+	if err != nil {
+		return nil, err
+	}
+	if len(layers) != 1 {
+		return nil, errors.New("expected exactly one layer")
+	}
+	mediaType, err := layers[0].MediaType()
+	if err != nil {
+		return nil, err
+	}
+	if !strings.HasPrefix(string(mediaType), "application/vnd.dev.sigstore.bundle") {
```

Comment:
Should we be checking for specific bundle versions so that we don't attempt to unmarshal to an unsupported structure?

Reply:
The call to `b.UnmarshalJSON` below takes care of the version-specific validation. I also added a call to b.MinVersion to enforce bundle v0.3 or newer.